lms/product/factory.py

```diff
@@ -0,0 +1,39 @@
+# We need to import all classes so we are aware of them for the mapping below
+from lms.product.blackboard import BlackboardProduct  # pylint: disable=unused-import
+from lms.product.canvas import CanvasProduct  # pylint: disable=unused-import
+from lms.product.generic import GenericProduct
+from lms.product.product import Product
+
+
+def get_product_from_request(request) -> Product:
+    """Get the correct product object from the provided request."""
+
+    family = _get_family(request)
+    product_class = _get_implementing_class(family)
+    product = product_class.from_request(request)
+    product.family = family
+    return product
+
+
+def _get_implementing_class(family):
+    for lms_class in Product.__subclasses__():
```

Comment:
If we have to import them anyway I'd would have writing this just a dictionary {family -> class}.get(family, GenericProduct)

Nothing wrong with this approach in any case.

Reply:
I think that's significantly less fancy, in a good way. The auto discovery is prone to error or at least confusion.